timesketch/lib/tasks.py

```diff
@@ -96,76 +103,55 @@ def run_plaso(source_file_path, timeline_name, index_name, username=None):
 
 
 @celery.task(track_started=True)
-def run_csv(source_file_path, timeline_name, index_name, username=None):
-    """Create a Celery task for processing a CSV file.
+def run_csv_jsonl(source_file_path, timeline_name, index_name, source_type,
+                  username=None):
+    """Create a Celery task for processing a CSV or JSONL file.
 
     Args:
-        source_file_path: Path to CSV file.
+        source_file_path: Path to CSV or JSONL file.
         timeline_name: Name of the Timesketch timeline.
         index_name: Name of the datastore index.
+        source_type: Type of file, csv or jsonl.
         username: Username of the user who will own the timeline.
 
     Returns:
         Dictionary with count of processed events.
     """
     event_type = u'generic_event'  # Document type for Elasticsearch
+    validators = {
+        u'csv': read_and_validate_csv,
+        u'jsonl': read_and_validate_jsonl
+    }
+    read_and_validate = validators.get(source_type)
 
     # Log information to Celery
     logging.info(u'Index name: %s', index_name)
     logging.info(u'Timeline name: %s', timeline_name)
+    logging.info(u'Source type: %s', source_type)
     logging.info(u'Document type: %s', event_type)
     logging.info(u'Owner: %s', username)
 
     es = ElasticsearchDataStore(
         host=current_app.config[u'ELASTIC_HOST'],
         port=current_app.config[u'ELASTIC_PORT'])
 
-    es.create_index(index_name=index_name, doc_type=event_type)
-    for event in read_and_validate_csv(source_file_path):
-        es.import_event(index_name, event_type, event)
-
-    # Import the remaining events
-    total_events = es.import_event(index_name, event_type)
+    # Reason for the broad exception catch is that we want to capture
+    # all possible errors and exit the task.
+    try:
+        es.create_index(index_name=index_name, doc_type=event_type)
+        for event in read_and_validate(source_file_path):
+            es.import_event(index_name, event_type, event)
+        # Import the remaining events
+        total_events = es.import_event(index_name, event_type)
+    except Exception as e:
```

Comment:
todo (for future release): catch specific errors and handle.

Reply:
Reason for the broad exception catch is that we want to capture
 -    for event in read_and_validate_csv(source_file_path):		 +    # all possible errors and exit the task.